incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -235,6 +236,8 @@ private MqttSubscribeProxy(
             this.retainAsPublished = new Long2ObjectHashMap<>();
             this.messages = new KafkaMessagesProxy(originId, resolvedId, this);
             this.retained = new KafkaRetainedProxy(originId, resolvedId, this);
+            this.kafkaMessagesTopic = kafkaMessagesTopic;
+            this.kafkaRetainedTopic = kafkaRetainedTopic;
```

Comment:
Perhaps we should pass these topic names into the `KafkaMessagesProxy` and `KafkaRetainProxy`, then just call the field `topic` and avoid passing it to methods on the `KafkaXxxProxy` classes?